Allow zero sized dimensions in padding operations

sladyn98 opened the pull request:
Previously, the padding implementation in PadNd.cpp required all output dimensions to be strictly positive (> 0), which caused errors when padding tensors with zero-sized dimensions even when the padding for that dimension was also zero.

This change relaxes the constraint to allow non-negative (>= 0) output dimensions, enabling operations like:
  x = torch.ones((0, 1))
  y = functional.pad(x, [1, 1, 0, 0])  # Now returns tensor of shape (0, 3)

The previous behavior was unnecessarily restrictive since mathematically a zero-sized dimension with zero padding should remain zero-sized.

Fixes https://github.com/pytorch/pytorch/issues/152750

sladyn98 commented:
@pytorchbot label "topic: not user facing"

albanD reviewed (commented):
Does the kernel properly handle 0-dim Tensors below?
This would need appropriate testing. Most likely in "common_method_invocation.py" in the sample input method for the padding function you care about in python.

sladyn98 commented:
@albanD  How do i verify if the kernel handles this ?  this is my first contribution so would like to learn more about it.

albanD commented:
You have to read the code of the kernel haha I usually assume that if the assert is there, it's because something below it makes that assumption.
The other way is to add test and see if they pass haha

sladyn98 committed:
Fix torch==2.6 broke nn.Module.dtype typing - Issue #152292

- Add explicit device and dtype properties to nn.Module class
- Fixes typing issue where module.device and module.dtype were incorrectly typed
- Properties return torch.device and torch.dtype respectively
- Added comprehensive tests for the new properties

Fixes #152292

sladyn98 committed:
Merge branch 'fix-issue-152292-attempt-1' of github.com:sladyn98/pytorch

sladyn98 committed:
Allow zero-sized dimensions in padding operations

Previously, the padding implementation in PadNd.cpp required all output
dimensions to be strictly positive (> 0), which caused errors when padding
tensors with zero-sized dimensions even when the padding was zero.

This commit changes the check from > 0 to >= 0, allowing mathematically
valid operations like padding a tensor of shape (0, 1) to (0, 3).

Added test cases for both 0-dimensional tensors and tensors with
zero-sized dimensions to ensure proper coverage.

albanD commented on torch/nn/modules/module.py:
```diff
@@ -1943,6 +1943,52 @@ def __setstate__(self, state):
             self._backward_pre_hooks = OrderedDict()
 
+    @property
+    def device(self) -> device:
```
This is very unrelated to the change advertized on the title and description!
